client/resource_group/controller/controller.go

```diff
@@ -345,6 +349,14 @@ func (c *ResourceGroupsController) updateAvgRequestResourcePerSec() {
 	})
 }
 
+func (c *ResourceGroupsController) resetEmergencyTokenAcquisition() {
+	c.groupsController.Range(func(name, value any) bool {
+		gc := value.(*groupCostController)
+		gc.resetEmergencyTokenAcquisition()
+		return true
+	})
+}
+
```

Comment:
for all including `c.groupsController.Range` funcs
Do you think it needs to abstract out all the traverse ranges and pass in specific functions each time?

Reply:
Good idea. We can impl it in another PR

Reply:
Ok, I will try.